diff --git a/Makefile b/Makefile
index 886b42f..573420c 100644
--- a/Makefile
+++ b/Makefile
@@ -869,7 +869,7 @@ install-includedir:
 	$(INSTALL_DIR) "$(DESTDIR)$(includedir)"
 
 install: all $(if $(BUILD_DOCS),install-doc) \
-	install-datadir install-localstatedir install-includedir \
+	install-datadir install-includedir \
 	$(if $(INSTALL_BLOBS),$(edk2-decompressed)) \
 	recurse-install
 ifneq ($(TOOLS),)
