diff --git a/Makefile b/Makefile
index f78a1b0..7cc9945 100644
--- a/Makefile
+++ b/Makefile
@@ -193,7 +193,7 @@ META_TGT_WHITELIST+= \
 
 PATH=	/sbin:/bin:/usr/sbin:/usr/bin
 MAKEOBJDIRPREFIX?=	/usr/obj
-_MAKEOBJDIRPREFIX!= /usr/bin/env -i PATH=${PATH} ${MAKE} MK_AUTO_OBJ=no \
+_MAKEOBJDIRPREFIX!= env -i PATH=${PATH} ${MAKE} MK_AUTO_OBJ=no \
     ${.MAKEFLAGS:MMAKEOBJDIRPREFIX=*} __MAKE_CONF=${__MAKE_CONF} \
     SRCCONF=${SRCCONF} SRC_ENV_CONF= \
     -f /dev/null -V MAKEOBJDIRPREFIX dummy
diff --git a/sys/conf/newvers.sh b/sys/conf/newvers.sh
index 2aa79b4..4ab4769 100644
--- a/sys/conf/newvers.sh
+++ b/sys/conf/newvers.sh
@@ -196,7 +196,7 @@ u=${USER:-root}
 d=$(pwd)
 h=${HOSTNAME:-$(hostname)}
 if [ -n "$SOURCE_DATE_EPOCH" ]; then
-	if ! t=$(date -r $SOURCE_DATE_EPOCH 2>/dev/null); then
+	if ! t=$(date -d @$SOURCE_DATE_EPOCH 2>/dev/null); then
 		echo "Invalid SOURCE_DATE_EPOCH" >&2
 		exit 1
 	fi
diff --git a/tools/build/Makefile b/tools/build/Makefile
index a0026c5..5b9cb43 100644
--- a/tools/build/Makefile
+++ b/tools/build/Makefile
@@ -194,7 +194,7 @@ _host_tools_to_symlink+=xcrun
 .endif  # ${.MAKE.OS} == "Darwin"
 # On Ubuntu /bin/sh is dash which is totally useless. Let's just link bash
 # as the build sh since that will work fine.
-_host_abs_tools_to_symlink+=	/bin/bash:sh
+_host_abs_tools_to_symlink+=	${bash}/bin/bash:sh
 _host_tools_to_symlink:=${_host_tools_to_symlink:Nsh}
 .endif
 
diff --git a/tools/build/mk/Makefile.boot b/tools/build/mk/Makefile.boot
index 74f8161..2775410 100644
--- a/tools/build/mk/Makefile.boot
+++ b/tools/build/mk/Makefile.boot
@@ -15,9 +15,9 @@ BUILD_TOOLS_LDFLAGS=${LDFLAGS}
 # nm may not work as expected if the nm for the target architecture comes in
 # $PATH before a nm that supports the host architecture.
 # To ensure that host binary compile as expected we use the tools from /usr/bin.
-AR:=	/usr/bin/ar
-RANLIB:=	/usr/bin/ranlib
-NM:=	/usr/bin/nm
+AR:=	ar
+RANLIB:=	ranlib
+NM:=	nm
 
 # Don't use lorder and tsort since lorder is not installed by default on most
 # Linux systems and the FreeBSD lorder does not work on Linux. For the bootstrap
