diff --git a/debian/setup_chroot.sh b/debian/setup_chroot.sh
index 9d39a70..1b80dd8 100755
--- a/debian/setup_chroot.sh
+++ b/debian/setup_chroot.sh
@@ -8,10 +8,7 @@ set -e
 debian_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
 setup_scripts_dir="${debian_dir}/setup_scripts"
 
-# NOTE: not excluding docs, because the script fails to build as such
-# As a result we have a larger binary that might not fit into flash
-#"${setup_scripts_dir}/exclude_docs.sh"
-echo "Skipping exclude_docs.sh - this might lead to a larger binary"
+"${setup_scripts_dir}/exclude_docs.sh"
 "${setup_scripts_dir}/install_important.sh"
 
 # Install an init system.
diff --git a/debian/setup_scripts/exclude_docs.sh b/debian/setup_scripts/exclude_docs.sh
index 4f4dfd7..41e9c2c 100755
--- a/debian/setup_scripts/exclude_docs.sh
+++ b/debian/setup_scripts/exclude_docs.sh
@@ -5,10 +5,12 @@ set -e
 echo "
 path-exclude=/usr/share/man/*
 path-exclude=/usr/share/locale/*
-path-include=/usr/share/locale/locale.alias
 path-exclude=/usr/share/doc/*
 path-include=/usr/share/doc/*/copyright
 path-exclude=/usr/share/dict/*
+path-include=/usr/share/locale/locale.alias
+path-include=/usr/share/locale/en/*
+path-include=/usr/share/locale/en_US.UTF-8/*
 " >/etc/dpkg/dpkg.cfg.d/exclude
 
 # Reinstall all packages to make it take effect
diff --git a/debian/setup_scripts/install_important.sh b/debian/setup_scripts/install_important.sh
index 5ce5959..2502ce0 100755
--- a/debian/setup_scripts/install_important.sh
+++ b/debian/setup_scripts/install_important.sh
@@ -26,4 +26,5 @@
 apt install -y usbutils
 
 # (not needed yet) set of tools for manipulating NVMe drives
-# apt-get install -y nvme-cli
\ No newline at end of file
+# apt-get install -y nvme-cli
+apt-get remove -y rsyslog"
\ No newline at end of file

